src/Raven.Studio/typescript/components/pages/database/tasks/ongoingTasks/editTasks/editGenAiTask/partials/steps/EditGenAiTaskStepBasic.tsx

```diff
@@ -0,0 +1,131 @@
+    const databaseName = useAppSelector(databaseSelectors.activeDatabaseName);
+
+    const handleTest = async () => {
+        const connectionString = aiConnectionStrings[formValues.connectionStringName];
+
+        dispatch(
+            editGenAiTaskActions.testConnectionString({
+                databaseName,
+                connectorType: getConnectorType(connectionString),
+                settings: mapAiConnectionStringToSettingsDto(connectionString),
+            })
+        );
+    };
+
+    const handleNext = async () => {
+        const isValid = await trigger(["name", "state", "responsibleNode", "connectionStringName"]);
+
+        if (isValid) {
+            dispatch(editGenAiTaskActions.currentStepSet("context"));
+        }
+    };
+
+    return (
+        <HStack gap={2} className="justify-content-end">
+            <ButtonWithSpinner
+                variant="info"
+                className="rounded-pill"
+                onClick={handleTest}
+                isSpinning={connectionStringTest.status === "loading"}
+                icon="test"
+            >
+                Test connection
+            </ButtonWithSpinner>
+
+            <Button variant="primary" className="rounded-pill" onClick={handleNext}>
+                Next <Icon icon="arrow-right" margin="ms-1" />
+            </Button>
+        </HStack>
+    );
+}
+
+const getConnectorType = (
+    connection: Raven.Client.Documents.Operations.AI.AiConnectionString
+): Raven.Client.Documents.Operations.AI.AiConnectorType => {
+    if (connection.AzureOpenAiSettings) {
+        return "AzureOpenAi";
+    }
+    if (connection.GoogleSettings) {
+        return "Google";
+    }
+    if (connection.HuggingFaceSettings) {
+        return "HuggingFace";
+    }
+    if (connection.OllamaSettings) {
+        return "Ollama";
+    }
+    if (connection.EmbeddedSettings) {
+        return "Embedded";
+    }
+    if (connection.OpenAiSettings) {
+        return "OpenAi";
+    }
+    if (connection.MistralAiSettings) {
+        return "MistralAi";
+    }
+
+    return "None";
+};
+
+export function mapAiConnectionStringToSettingsDto(
+    connection: Raven.Client.Documents.Operations.AI.AiConnectionString
+): AiConnectionStringsSettings {
+    if (connection.AzureOpenAiSettings) {
+        return connection.AzureOpenAiSettings;
+    }
+    if (connection.GoogleSettings) {
+        return connection.GoogleSettings;
+    }
+    if (connection.HuggingFaceSettings) {
+        return connection.HuggingFaceSettings;
+    }
+    if (connection.OllamaSettings) {
+        return connection.OllamaSettings;
+    }
+    if (connection.EmbeddedSettings) {
+        return connection.EmbeddedSettings;
+    }
+    if (connection.OpenAiSettings) {
+        return connection.OpenAiSettings;
+    }
+    if (connection.MistralAiSettings) {
+        return connection.MistralAiSettings;
+    }
+
+    return null;
```

Comment:
Perhaps it'd be cleaner to implement `mapAiConnectionStringToSettingsDto` like tihs:

```suggestion
return [
  connection.AzureOpenAiSettings,
  connection.GoogleSettings,
  connection.HuggingFaceSettings,
  connection.OllamaSettings,
  connection.EmbeddedSettings,
  connection.OpenAiSettings,
  connection.MistralAiSettings
].find(Boolean) ?? null;
```